src/main/kotlin/org/jitsi/nlj/dtls/DtlsStack.kt

```diff
@@ -182,6 +182,22 @@ class DtlsStack(
         dtlsTransport?.send(data, off, len)
     }
 
+    private fun processIncomingProtocolData() {
+        var bytesReceived: Int
+        do {
+            synchronized(this) {
```

Comment:
I think the `synchronized` block should extend to include the arraycopy below since it uses `dtlsAppDataBuf`.

Also, maybe lock on `dtlsAppDataBuf` itself instead of the `DtlsStack` instance.